### PR TITLE
feat!: provision using a proof

### PR DESCRIPTION
**Pull request by Gozala**
Adds a couple of more commands

`w3 coupon create did:...` - That can be used to create delegation and pack it as a redeemable coupon.
`w3 space provision --coupon https://gozala.io/coupon` - That can be used to provision space with pre-arranged coupon
`w3 plan get` - Prints current plan info

All the above are in support of the workshop and specifically so we could create short lived coupon for workshop participants that would allow them to provision spaces without signing up as customer.

### CODE REVIEW

File: test/bin.spec.js [hunk 1]
```diff
@@ -514,6 +519,41 @@ export const testSpace = {
       'added provider shows up in the space info'
     )
   }),
+
+  'only w3 space provision': test(async (assert, context) => {
```
Comment (olizilla) [inline test/bin.spec.js:523]: i dont know how entail works but `only` is probably not what we want here

Comment (Gozala) [inline test/bin.spec.js:523]:
```suggestion
  'w3 space provision': test(async (assert, context) => {
```

**Comment by olizilla** [comment]
Is the intended usecase here for ephemeral environments? can you add more context to the description here? Are you imagining folks running a services that basically do this

https://github.com/web3-storage/w3cli/pull/123/files#diff-c22c0a8f2ed21fa3523fdd0f0072318b72393986fc13ea1d30aabb2f721fe335R1225-R1271

that their CI environments or untrusted apps can call to get a short lived delegation?

**Review by alanshaw — APPROVED** [review]
LGTM

**Pull request merged**